### ISSUE TITLE
PEP 3127 Remove old syntax for octal literals

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8347]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)| @rodrigc reported|
|-|-|
|Trac ID|trac#8347|
|Type|enhancement|
|Created|2016-05-22 11:36:34Z|
|Branch|https://github.com/twisted/twisted/tree/8347-rodrigc-octal-2|

PEP 3127 ( ​​https://www.python.org/dev/peps/pep-3127/ ) deprecates the old representation of octal literals.

The old way of representing an octal literal was by prepending with a **0**

```
a = 0777
```

The new way is to prepend with **0o**

```
a = 0o777
```

In Python 2.6 and Python 2.7, both methods work, but the old way is deprecated.

In Python 3.x, only the new way works.  The old way results in an error.

<details><summary>Searchable metadata</summary>

```
trac-id__8347 8347
type__enhancement enhancement
reporter__rodrigc rodrigc
priority__normal normal
milestone__Python_3_x Python-3.x
branch__8347_rodrigc_octal_2 8347-rodrigc-octal-2
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1463916994880024 1463916994880024
changetime__1465255725408715 1465255725408715
version__None None
owner__rodrigc rodrigc

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_1"></a>|@rodrigc commented|
|-|-|

See:
https://github.com/twisted/twisted/pull/68

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_2"></a>|@adiroiban set owner to @rodrigc|
|-|-|

Changes looks good. Thanks.

I have pushed a small commit in which I am also updating the octal representation from the man page... for consistency :)

The changes are not that big and I think that is OK to have them outside of a ticket which only convert them for a specific module....and when a ticket is porting a module, the diff should be more focused on specific py3 issues

I am fighting the buildbot builds ... if this is not merged by the time you read this, please ping me

thanks again!

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_3"></a>|@adiroiban commented|
|-|-|

Finally all builders are green.

I will add the topfile news fragment, so that this can be merged.

Will merge. Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_4"></a>|@adiroiban set status to `closed`|
|-|-|

In changeset 8bef76bf809fa729fc0a66e56865f1e5bb9eaa18
```
#!CommitTicketReference repository="" revision="8bef76bf809fa729fc0a66e56865f1e5bb9eaa18"
Merge rodrigc-pep3127_octal-8347: Use new syntax for octal literals.

Author: rodrigc
Reviewer: adiroiban
Fixes: #8347
```

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_5"></a>|@hawkowl commented|
|-|-|

In changeset 3c9b4b3c5f7e50da87732a1ed9707a0163c938aa
```
#!CommitTicketReference repository="" revision="3c9b4b3c5f7e50da87732a1ed9707a0163c938aa"
Revert "Merge rodrigc-pep3127_octal-8347: Use new syntax for octal literals."

A wrong topfile was added. This reverts commit 8bef76bf809fa729fc0a66e56865f1e5bb9eaa18.

Reopens: #8347
```

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_6"></a>|@hawkowl commented|
|-|-|

In changeset 77d338bc63218f86eb3860d37653323d91c2fce7
```
#!CommitTicketReference repository="" revision="77d338bc63218f86eb3860d37653323d91c2fce7"
Merge rodrigc-pep3127_octal-8347: Use new syntax for octal literals.

Author: rodrigc
Reviewer: adiroiban
Fixes: #8347
```

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_7"></a>|@rodrigc set status to `reopened`|
|-|-|

This wasn't merged properly:
https://github.com/twisted/twisted/commit/77d338bc63218f86eb3860d37653323d91c2fce7

Please look again.  I submitted a new pull request:

https://github.com/twisted/twisted/pull/143

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)|@rodrigc removed owner<br>@rodrigc set status to `new`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_9"></a>|@adiroiban set owner to @rodrigc|
|-|-|

creating a new branch and sending to buildbot.

will merge if all is ok.

many thanks!

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_10"></a>|@adiroiban set status to `closed`|
|-|-|

In changeset dae9412b22e7434b0a117809d2f003fe8250e354
```
#!CommitTicketReference repository="" revision="dae9412b22e7434b0a117809d2f003fe8250e354"
Merge 8347-rodrigc-octal-2: Prefix octal literals with 0o.

Author: rodrigc
Reviewer: adiroiban
Fixes: #8347
```
